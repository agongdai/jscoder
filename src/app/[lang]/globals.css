@tailwind base;
@tailwind components;
@tailwind utilities;

html {
  font-size: 62.5%;
  scroll-behavior: smooth;
}

body {
  @apply m-0 overflow-y-auto overflow-x-hidden text-base antialiased font-medium;
}

html,
body,
main {
  @apply bg-bg-light-main box-border h-full overflow-y-auto text-text-primary;
}

html.dark,
html.dark body,
html.dark main {
  @apply bg-bg-dark-main text-white;
}

.jsc-scrollbar {
  /** // Firefox - Gecko **/
  scrollbar-width: thin;
  scrollbar-color: rgba(42, 42, 42, 0.2) rgba(93, 93, 93, 0.05);

  /** // IE >= 8 **/
  scrollbar-track-color: rgba(93, 93, 93, 0.05);
  scrollbar-face-color: rgba(42, 42, 42, 0.2);
  width: calc(100% - 0.6rem);

  @apply relative right-1 left-0 overflow-y-auto max-h-full;
}

.jsc-scrollbar::-webkit-scrollbar {
  @apply visible w-[0.6rem] h-[0.6rem];
}

.jsc-scrollbar::-webkit-scrollbar-track {
  @apply rounded bg-gray-400;
}

.jsc-scrollbar::-webkit-scrollbar-thumb {
  @apply rounded bg-gray-600;
}
