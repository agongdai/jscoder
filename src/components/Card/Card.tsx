import React from 'react';
import cx from 'classnames';

interface Props {
  label: string;
  children: React.ReactNode;
  className?: string;
}

export default function Card({ label, children, className = '' }: Props) {
  return (
    <div
      className={cx(
        'shadow rounded-2xl bg-bg-light-light dark:bg-bg-dark-light relative min-h-[16rem]',
        className,
      )}
    >
      <div
        role='heading'
        aria-level={5}
        className='absolute left-0 top-0 bg-primary-main text-white text-sm font-semibold py-1 px-2 rounded-ss-2xl rounded-ee-2xl'
      >
        {label}
      </div>
      <div className='text-6xl font-medium justify-center flex items-center h-full w-full flex-wrap'>
        {children}
      </div>
    </div>
  );
}
