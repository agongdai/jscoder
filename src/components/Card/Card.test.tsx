import { expect } from '@jest/globals';
import { render, screen } from '@testing-library/react';

import Card from './Card';

describe('Home', () => {
  it('renders a heading', () => {
    const { container } = render(<Card label='Card Header'>whatever</Card>);

    const heading = screen.getByRole('heading', {
      name: /Card Header/i,
    });

    expect(heading).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
