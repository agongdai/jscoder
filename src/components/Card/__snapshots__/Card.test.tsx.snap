// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders a heading 1`] = `
<div>
  <div
    class="shadow rounded-2xl bg-bg-light-light dark:bg-bg-dark-light relative min-h-[16rem]"
  >
    <div
      aria-level="5"
      class="absolute left-0 top-0 bg-primary-main text-white text-sm font-semibold py-1 px-2 rounded-ss-2xl rounded-ee-2xl"
      role="heading"
    >
      Card Header
    </div>
    <div
      class="text-6xl font-medium justify-center flex items-center h-full w-full flex-wrap"
    >
      whatever
    </div>
  </div>
</div>
`;
